internal/equality/equality.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package equality
+
+import (
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	gatewayapi_v1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+)
+
+// GatewayClassStatusChanged checks if current and expected match and if not,
+// returns true.
+func GatewayClassStatusChanged(current, expected gatewayapi_v1alpha1.GatewayClassStatus) bool {
+	return !apiequality.Semantic.DeepEqual(current.Conditions, expected.Conditions)
```

Comment:
We have `conditionChanged` below, could this use that to verify if something changed? Seems like we now have 2 different ways of doing comparisons.

Reply:
I created the equality package since we'll have to perform current/desired equality comparisons for different objects, fields, etc. ([xref](https://github.com/projectcontour/contour-operator/blob/main/internal/equality/equality.go)). `conditionChanged` is meant to compare 2 conditions, whereas `GatewayClassStatusChanged` is meant to compare 2 condition slices. I'll update the PR to add a private func in the status pkg to compare the condition slice of current/desired gatewayclasses.